src/main/java/uk/gov/hmcts/dm/config/batch/OrphanDocumentDeletionTask.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.dm.config.batch;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import net.javacrumbs.shedlock.spring.annotation.EnableSchedulerLock;
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dm.domain.StoredDocument;
+import uk.gov.hmcts.dm.service.StoredDocumentService;
+import uk.gov.hmcts.dm.service.batch.AuditedStoredDocumentBatchOperationsService;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This task periodically checks for CSV files in the orphandocuments blob container.
+ * If it finds one it will download
+ * it and then delete all the documents found with the UUID provided in the file.
+ * After the deletion has been completed the file
+ * is removed from the blob container.
+ */
+
+@Service
+@ConditionalOnProperty("toggle.orphandocumentdeletion")
+@EnableScheduling
+@EnableSchedulerLock(defaultLockAtMostFor = "PT5M")
+public class OrphanDocumentDeletionTask {
+    private static final Logger log = LoggerFactory.getLogger(OrphanDocumentDeletionTask.class);
+    private static final String TASK_NAME = "Orphan-Document-Deletion-Task";
+
+    private final BlobContainerClient blobClient;
+    private final StoredDocumentService documentService;
+    private final AuditedStoredDocumentBatchOperationsService auditedStoredDocumentBatchOperationsService;
+
+    public OrphanDocumentDeletionTask(
+        @Qualifier("orphandocument-storage") BlobContainerClient blobClient,
+        StoredDocumentService documentService,
+        AuditedStoredDocumentBatchOperationsService auditedStoredDocumentBatchOperationsService
+    ) {
+        this.blobClient = blobClient;
+        this.documentService = documentService;
+        this.auditedStoredDocumentBatchOperationsService = auditedStoredDocumentBatchOperationsService;
+    }
+
+    @Scheduled(cron = "${spring.batch.orphanFileDeletionCronJobSchedule}")
+    @SchedulerLock(name = TASK_NAME)
```

Comment:
Can you please env in the Task name as done here - https://github.com/hmcts/document-management-store-app/blob/master/src/main/java/uk/gov/hmcts/dm/config/batch/MetadataMigrationConfiguration.java#L61C36-L61C39.  ?
This will help ensure no job is been run across environments.